wordpress-mobile/WordPress-Android
# Adding Tracks Event for Suggestions UI

This adds a track event for usage of the suggestions UI.

### To Test

Make sure you have `Me -> App Settings -> Privacy Settings -> Collect Information`. Personally, when working with tracks I like to leave that unchecked and spend 30 minutes debugging the code to figure out why tracks aren't working before I realize that is unchecked, but I understand if you want to skip straight to flipping the toggle 😉 🤣 .

All of the tests below should be done with both (a) xposts and (b) user-mentions. In _all_ the tests, the event should have a `suggestion_type` property with a value of either "xpost" or "user" as appropriate.

This event still needs to be marked as valid in the Tracks directory. Please either do that as part of reviewing this PR, or remind me that it still needs to be done, and I'm happy to take care of it once this PR is approved. 🙇 

Related: p9ugOq-1tR-p2

#### 1. Successfully select a suggestion
1. Select a P2 site
1. Open the suggestions UI by typing a `+`/`@` or tapping the suggestions button on the format bar.
1. Select a suggestion by either (a) tapping on one of the available suggestions, or (b) filtering the list to a single suggestion and pressing enter
1. Observe that the appropriate track event ("suggestion_session_finished") is sent with `did_select_suggestion` as `true` and an appropriate `suggestion_type`.

#### 2. Exiting without selecting a suggestion
1. Select a P2 site
1. Open the suggestions UI by typing a `+`/`@` or tapping the suggestions button on the format bar.
1. Exit the suggestions UI by either
    1. Pressing backspace before entering any text in the suggestion UI;
    2. Filtering the list of suggestions so it does not take up the full screen and tapping "behind" the suggestions UI (i.e., on the block editor you can see in the background); or
    3. Using the back button/swipe. Note that if the keyboard is shown you will probably have to swipe back two times (first time closes the keyboard and second time exits the suggestions UI), but an event should only be sent when you exit the suggestions UI.
1. Observe that the appropriate track event ("suggestion_session_finished") is sent with `did_select_suggestion` as `false` and an appropriate `suggestion_type`.

PR submission checklist:
